### PR TITLE
making clearing method clear the history element

### PR DESCRIPTION
**Pull request by yaaHosein**
Let us work together!!

### CODE REVIEW

**Comment by SMH110** [comment]
I open the files changed tab and this what I saw
image


Can you please tell me why you have this much of code being edited while you just added one method?

**Comment by yaaHosein** [comment]
Very strange
I did not touched these lines last night
I just added maby only one line inside clearing method and uncomment this method

**Comment by yaaHosein** [comment]
> I open the files changed tab and this what I saw
> image
> 
> Can you please tell me why you have this much of code being edited while you just added one method?

Very strange
I did not touched these lines last night
I just added only one or 2 line inside clearing method and uncomment this method

**Comment by yaaHosein** [comment]
> I open the files changed tab and this what I saw
> image
> 
> Can you please tell me why you have this much of code being edited while you just added one method?

Very strange
I did not touched these lines last night
I just added maby only one line inside clearing method and uncomment this method

**Comment by SMH110** [comment]
Can you please tell exactly what you did after completing the PR ?

**Comment by yaaHosein** [comment]
> Can you please tell exactly what you did after completing the PR ?

As you are aware I had added clearing method a long time ago
So just  uncommented it
And added 2 line
Assigning history to ""
And updating the history

I did the changes and checking first in a different file (draft file( 
I do not remember if by mistake touched anything in the orginal file

Maybe these changes are very old changes
But I do not know if they are old why it dosent pushed eralier

**Comment by SMH110** [comment]
> > Can you please tell exactly what you did after completing the PR ?
> 
> As you are aware I had added clearing method a long time ago
> So just uncommented it
> And added 2 line
> Assigning history to ""
> And updating the history
> 
> I did the changes and checking first in a different file (draft file(
> I do not remember if by mistake touched anything in the orginal file
> 
> Maybe these changes are very old changes
> But I do not know if they are old why it dosent pushed eralier

I mean how did you create the new branch 
did you pull your changes from github ..etc

**Comment by yaaHosein** [comment]
Yes I did
First checkout to master
Then git pull
Then creating the new branch and pushing it
Then pushing new changes 
Then pull request

**Comment by SMH110** [comment]
You have removed some code which exist in master 
image

I have opened your master branch (M) next to the feature branch (F)
As part of your changes you removed some code was on the master branch.

File: DevelopedCalculator.js [hunk 1]
```diff
@@ -19,17 +19,19 @@ class DevelopedCalculator extends Calculator {
...
+        super.onOperatorButtonClick(event);
+        // pushing current result and the operator to the history array
+        this.history.push(this.currentResult, this.operator);
+        // calling gettingHistoryArray to update the history element after clicking any operator button 
+        this.gettingHistoryArray();
+        // this.history.push(this.currentResult, this.operator);
+    }
+    onClearButtonClick() {
+        super.onClearButtonClick()
+        this.history = ""
```
Comment (SMH110) [inline DevelopedCalculator.js]: is the history array or string?

**Comment by yaaHosein** [comment]
Just I check these when I was near the laptp
I just discommented the lines which you drew a circle around them and added to more lines of code 
I do not know why this is happened
Should I checkout to the master and git pull?

**Comment by yaaHosein** [comment]
Array

File: DevelopedCalculator.js (hunk 1 above)
Comment (yaaHosein) [reply DevelopedCalculator.js]: Array

**Comment by yaaHosein** [comment]
I do not see anything wrong in master or feature branches
What you draw a circle around it was before and now it is on the baranch

I added line 32 and 33 in the new featur branch

File: DevelopedCalculator.js (hunk 1 above)
Comment (yaaHosein) [reply DevelopedCalculator.js]: I can see what I did is working even if the history is an array

Comment (yaaHosein) [reply DevelopedCalculator.js]: I can see what I did is working even if the history is an array

Comment (SMH110) [reply DevelopedCalculator.js]: if it's array, then use an empty array instead of empty string

File: DevelopedCalculator.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ class DevelopedCalculator extends Calculator {
     }
     onClearButtonClick() {
         super.onClearButtonClick()
-        this.history = ""
+        this.history = [""]
```
Comment (SMH110) [inline DevelopedCalculator.js]: ` [""]` is this empty array?

File: DevelopedCalculator.js [hunk 3]
```diff
@@ -19,17 +19,19 @@ class DevelopedCalculator extends Calculator {
...
+        // pushing current result and the operator to the history array
+        this.history.push(this.currentResult, this.operator);
+        // calling gettingHistoryArray to update the history element after clicking any operator button 
+        this.gettingHistoryArray();
+        // this.history.push(this.currentResult, this.operator);
+    }
+    onClearButtonClick() {
+        super.onClearButtonClick()
+        this.history = [""]
+        this.historyElement.textContent = this.history
```
Comment (SMH110) [inline DevelopedCalculator.js]: what will happen if you call `this.gettingHistoryArray();` instad of `this.historyElement.textContent = this.history`

**Comment by SMH110** [comment]
Have you added a comment to your changes? will you remember the role of "onClearButtonClick" in the future?

**Comment by SMH110** [comment]
you can complete this pr

File: DevelopedCalculator.js (hunk 2 above)
Comment (yaaHosein) [reply DevelopedCalculator.js]: I have pushed the correct one before reading this comment

**Pull request merged**